src/app/carbon-estimation-table/carbon-estimation-table.component.spec.ts

```diff
@@ -0,0 +1,357 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { CarbonEstimationTableComponent, TableItem } from './carbon-estimation-table.component';
+import { CarbonEstimationUtilService } from '../services/carbon-estimation-util.service';
+import { CarbonEstimation } from '../types/carbon-estimator';
+import { EmissionsLabels } from '../carbon-estimation/carbon-estimation.constants';
+
+describe('CarbonEstimationTableComponent', () => {
+  let component: CarbonEstimationTableComponent;
+  let fixture: ComponentFixture<CarbonEstimationTableComponent>;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [CarbonEstimationTableComponent],
+      providers: [CarbonEstimationUtilService],
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(CarbonEstimationTableComponent);
+    component = fixture.componentInstance;
+    const carbonEstimation: CarbonEstimation = {
+      version: '1.0',
+      upstreamEmissions: {
+        software: 7,
+        employee: 6,
+        network: 6,
+        server: 6,
+      },
+      directEmissions: {
+        employee: 9,
+        network: 8,
+        server: 8,
+      },
+      indirectEmissions: {
+        cloud: 9,
+        saas: 8,
+        managed: 8,
+      },
+      downstreamEmissions: {
+        endUser: 13,
+        networkTransfer: 12,
+      },
+    };
+
+    fixture.componentRef.setInput('carbonEstimation', carbonEstimation);
+    fixture.detectChanges();
+  });
+
+  it('should toggle display value of child emissions when toggle called', () => {
+    component.toggle(EmissionsLabels.Upstream);
+    fixture.detectChanges();
+    component.tableData().forEach(emission => {
+      if (emission.level === 1 && emission.category === EmissionsLabels.Upstream) {
+        expect(emission.expanded).toBeFalse();
+      }
+      if (emission.level === 2 && emission.parent === EmissionsLabels.Upstream) {
+        expect(emission.display).toBeFalse();
+      }
+    });
+  });
+
+  it('should set child emissions to display by default', () => {
+    component.tableData().forEach(emission => {
+      if (emission.level === 2 && emission.parent) {
+        expect(emission.display).toBeTrue();
+      }
+    });
+  });
+
+  it('should get emissions when getEmissions called', () => {
+    const emissions = component.getTableData(component.carbonEstimation());
```

Comment:
Seems like `getTableData()` is only exposed for this test, maybe it should be private and test the `tableData` instead?